src/postgres/src/interfaces/libpq/fe-connect.c

```diff
@@ -748,6 +810,611 @@ PQconnectStartParams(const char *const *keywords,
+	 * Try connecting with all the Private IP address.
+	 */
+	if (YBclientNetworkStatus==0)
+	{
+		int itr;
+		for(itr=0; itr<nServers; itr++)
+		{
+			if(YBtestNetwork(PQgetvalue(res, itr , 0)))
+			{
+				YBclientNetworkStatus=1;
+				break;
+			}
+		}
+	}
+
+	/*
+	 * Since the Client was unable to connect with the Private ips.
+	 * It must not have access to it. Thus we need to make the connection
+	 * to with the public ip.
+	 */
+	if (YBclientNetworkStatus==0)
+		YBclientNetworkStatus=-1;
+
+	/*
+	 * Locking the map_ready so that no other thread can start iterating it.
+	 */
+	pthread_mutex_lock(&(map_ready));
+
+	/*
+	 * Update the yb_last_update_time
+	 */
+	yb_last_update_time = time(NULL);
+
+	/*
+	 * 1. Assign the value of is_running as false for all the servers.
+	 * 2. Iterate the result and update the is_running as true.
+	 * 3. Maintain the count of the servers which were not found in the server_details.
+	 * 4. If this count is not zero reallocate the map memory and add the values to it.  
+	 */
+
+	/*
+	 * Assign the value of is_running as false for all the servers.
+	 */
+	for(int i =0; i< total_servers; i++)
+	{
+		server_details[i].is_running = false;
+	}
+
+	/*
+	 * Iterate the result and update the is_running as true.
+	 */
+  	for (int i = 0; i < nServers; i++)
+  	{
+	  
+		char  *server;
+		if(YBclientNetworkStatus==1) 
+			server= PQgetvalue(res, i , 0);	
+		else 
+			server= PQgetvalue(res, i , 7);
+		
+		/*
+		 * Since no other thread is iterating the server_details (map_ready is locked),
+		 * YBserverStatusChange can be called without considering the 
+		 * thread_safe (keeping the value of check_thread_safe as false).
+		 * If kept true will endup in a deadlock.
+		 */
+		if (!YBserverStatusChange(server,true,false))
+		{
+			/*
+		 	* Maintain the count of the servers which were not found in the map.
+		 	*/
+			increase_map_size++;
+			server_to_add[i] = true;
+		}else
+			server_to_add[i] = false;
+
+	}
+
+	/*
+	 * 4. If count is not zero reallocate the map memory and add the values to it.  
+	 */
+	if(increase_map_size != 0)
+	{
+		/*
+		 * Allocate the space
+		 */
+		struct node_details *temp_server_details = (struct node_details *) malloc ( (increase_map_size + total_servers) * sizeof (struct node_details));
+		for(int i =0; i < total_servers; i++)
+			temp_server_details[i] = server_details[i];
+
+		/*
+		 * Add the server details
+		 */
+		for(int i =0; i < nServers; i++)
+		{
+			if(!server_to_add[i])
+				continue;
+			
+			/* 
+			 * Store the ip address
+			 */
+			char  *temp_host_id;
+			if(YBclientNetworkStatus==1)
+				temp_host_id= PQgetvalue(res, i, 0);
+			else
+				temp_host_id= PQgetvalue(res, i, 7);
+				
+			temp_server_details[total_servers].host_ip = (char *)malloc(sizeof(char)*(strlen(temp_host_id)+1));
+			strcpy(temp_server_details[total_servers].host_ip ,  temp_host_id)	;	
+			
+			/*
+			 * Reset the connection count
+			 */
+			temp_server_details[total_servers].connections = 0;
```

Comment:
Why reset the connection count? What if the server has live connections?

Reply:
Here, new serves are added to the "server_details". 
Thus the values for the connection counts are set to "0".

Reply:
Ok, that's just for the new servers.